src/ChainSafe.GamingSdk.Gelato/Gelato.cs

```diff
@@ -0,0 +1,262 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using ChainSafe.GamingSDK.EVM.Web3.Core;
+using ChainSafe.GamingSDK.EVM.Web3.Core.Evm;
+using ChainSafe.GamingSdk.Gelato.Dto;
+using ChainSafe.GamingSdk.Gelato.Types;
+using ChainSafe.GamingWeb3;
+using ChainSafe.GamingWeb3.Environment;
+using Nethereum.ABI.EIP712;
+using Nethereum.Hex.HexTypes;
+using Web3Unity.Scripts.Library.Ethers.Contracts;
+using Web3Unity.Scripts.Library.Ethers.Providers;
+using Web3Unity.Scripts.Library.Ethers.Signers;
+
+namespace ChainSafe.GamingSdk.Gelato
+{
+    public class Gelato : IGelato, ILifecycleParticipant
+    {
+        private readonly GelatoClient gelatoClient;
+        private readonly IRpcProvider provider;
+        private readonly IContractBuilder contractBuilder;
+        private readonly ISigner signer;
+        private readonly GelatoConfig config;
+        private readonly IChainConfig chainConfig;
+
+        public Gelato(IHttpClient httpClient, IChainConfig chainConfig, GelatoConfig config, IRpcProvider provider, ISigner signer, IContractBuilder contractBuilder)
+        {
+            gelatoClient = new GelatoClient(httpClient, config);
+            this.provider = provider;
+            this.signer = signer;
+            this.config = config;
+            this.chainConfig = chainConfig;
+            this.contractBuilder = contractBuilder;
+        }
+
+        public async Task<RelayResponse> CallWithSyncFee(CallWithSyncFeeRequest request)
+        {
+            try
+            {
+                request.ChainId = int.Parse(chainConfig.ChainId);
+                return await gelatoClient.Post<CallWithSyncFeeRequest, RelayResponse>(RelayCall.CallWithSyncFee, request);
+            }
+            catch (Exception e)
+            {
+                throw new Web3Exception($"GelatoRelaySDK/relayWithSyncFee: Failed with error: {e.Message}");
+            }
+        }
+
+        public async Task<RelayResponse> CallWithSyncFeeErc2771(CallWithSyncFeeErc2771Request request)
+        {
+            return await CallWithSyncFeeErc2771(request, null);
+        }
+
+        // TODO: consume options
+        public async Task<RelayResponse> CallWithSyncFeeErc2771(CallWithSyncFeeErc2771Request request, RelayRequestOptions options)
+        {
+            if (signer == null)
+            {
+                throw new Web3Exception("GelatoRelaySDK/CallWithSyncFeeErc2771: No signer present for request");
```

Comment:
We could point out, that the signer should be registered during the build time, to make it more obvious.

Reply:
@oleksandrchainsafe The signer can only be present after a user connects their wallet, we aren't making builds with the players account in context

Reply:
Not sure what you mean